src/plan/markcompact/global.rs

```diff
@@ -44,6 +44,25 @@ pub const MARKCOMPACT_CONSTRAINTS: PlanConstraints = PlanConstraints {
     ..PlanConstraints::default()
 };
 
+/// A stage to compute forwarding addresses of objects. When we finish the marking transitive closure (`TRACE_KIND_MARK`),
+/// we calculate the forwarding addresses for marked objects.
+pub const CALCULATE_FORWARDING_STAGE: WorkBucketStage = WorkBucketStage::Custom(0);
+
+/// A stage to initiate the second transitive closure (`TRACE_KIND_FORWARD`). This happens after we calculate forwarding addresses.
+pub const SECOND_ROOTS_STAGE: WorkBucketStage = WorkBucketStage::Custom(1);
+
+/// A stage to forward weak references. This happens after the second transitive closure.
+pub const REF_FORWARDING_STAGE: WorkBucketStage = WorkBucketStage::Custom(2);
+
+/// A stage to forward finalizable objects. This happens after the second transitive closure.
+pub const FINALIZABLE_FORWARDING_STAGE: WorkBucketStage = WorkBucketStage::Custom(3);
+
+/// A stage to allow VM bindings to do ref forwarding. This happens after other ref forwarding work.
+pub const VM_REF_FORARDING_STAGE: WorkBucketStage = WorkBucketStage::Custom(4);
+
+/// A custom stage to do compaction. It happens after all the tracing stages, and before Release
+pub const COMPACT_STAGE: WorkBucketStage = WorkBucketStage::Custom(5);
```

Comment:
May organise them into an enum, such as
```
#[derive(enum_map::Enum)]
enum MarkCompactStage {
    CalculateForwarding,
    SecondRoots,
    RefForwarding,
    FinalizableForwarding,
    VMRefForwarding,
    Compact,
}
```
and automatically map them to and from `WorkBucketStage::Custom(...)`.  That should be simpler.